### DISCUSSION TITLE
Coral Inference speed slower than expected - not impressed

### DISCUSSION THREAD

**Discussion by stamandr** [blakeblackshear/frigate#5837] [Ask A Question]
I've read many post that users are getting between 5-10ms with Coral Accelerators.  I'm getting between 20-30 with the Coral USB and between 14-20 with the M.2 B+M key. 

The PC is a NUC6  (I5-6260u) with 8Gb Ram. The Coral USB is plugged into the USB 3 port. I'm running Frigate 0.11.1-2EADA21 in docker with compose ( recommended install method )

The M.2 B+M key was recently installed following the documentation https://coral.ai/docs/m2/get-started/#2a-on-linux/

My config for the 5 cameras are: Res: 704*480 , fps: 5

Everything works fine, It's just that the performance isn't close to what others have commented on.  Are there any special settings that need to be turned on/off in the BIOS to increase performance ? Any suggestions on what else I can try ?

**Comment by NickM-27** [comment]
There may be settings in the BIOS that helps. It is also possible that on your particular device multiple I/O share the same BUS (for example, some RPi devices have the ethernet and USB ports share the same BUS meaning neither hits full speed if both are used).

I can say that I hit 10 ms with USB and ~ 8 ms with m.2 and the default model (models do also affect the inference time)

**Reply by stamandr** [reply]
I agree that it is possible for the USB and ethernet sharing the same bus that could cause this, but that doesn't explain why the M.2 B+M key's inference speeds are so slow. I'm missing something here....

**Comment by kpcz** [comment]
For me Coral M2 A/E key never goes less than 10s.  It's usually about 10-13 ms. 10 cameras, various detection resolutions.
version 0.12b running in Unraid container.
I'm kinda happy with performance.

**Comment by burmjohn** [comment]
Coral USB via passthrough in a VM I'm getting 7.69 ms with 5 cameras. 4 cameras are at 720p and 1 is 1080p. I was getting about 10ms and some stuck detection issues until I passed through the entire hub.  It seems like you should be getting way better performance then what you are seeing. You are not virtualized are you?

**Reply by stamandr** [reply]
I running on a NUC6 (I5-6260u) with 8Gb Ram. The Coral USB is plugged into the USB 3 port. I'm running Frigate 0.11.1-2EADA21 in docker with compose ( recommended install method ). Barebones, not VM. This PC is more than adequate for Frigate
